src/openaq/incremental_downloader_parallel.py

```
@@ -74,45 +78,58 @@ def fetch_sensor_pages_parallel_sync(self, sensor_ids: List[int], max_pages_per_
                 all_data.extend(measurements)
-                
+                consecutive_errors = 0  # Reset error counter on success
+
                 if len(measurements) < 1000:
                     break
-                
+
                 page += 1
-                
+
             except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)